src/InstrumentationEngine/ProfilerManagerWrapper.cpp

```diff
@@ -0,0 +1,231 @@
+}
+
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::GetInstrumentationMethod(_In_ REFGUID cslid, _Out_ IUnknown** ppUnknown)
+{
+    HRESULT hr = S_OK;
+    CComPtr<IUnknown> pUnknown;
+    IfFailRet(m_pProfilerManager->GetInstrumentationMethod(cslid, &pUnknown));
+    *ppUnknown = pUnknown.Detach();
+    return S_OK;
+}
+
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::RemoveInstrumentationMethod(_In_ IInstrumentationMethod* pInstrumentationMethod)
+{
+    return m_pProfilerManager->RemoveInstrumentationMethod(pInstrumentationMethod);
+}
+
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::AddInstrumentationMethod(
+    _In_ BSTR bstrModulePath,
+    _In_ BSTR bstrName,
+    _In_ BSTR bstrDescription,
+    _In_ BSTR bstrModule,
+    _In_ BSTR bstrClassGuid,
+    _In_ DWORD dwPriority,
+    _Out_ IInstrumentationMethod** ppInstrumentationMethod)
+{
+    HRESULT hr = S_OK;
+    CComPtr<IInstrumentationMethod> pInstrumentationMethod;
+    IfFailRet(m_pProfilerManager->AddInstrumentationMethod(
+        bstrModulePath,
+        bstrName,
+        bstrDescription,
+        bstrModule,
+        bstrClassGuid,
+        dwPriority,
+        &pInstrumentationMethod));
+    *ppInstrumentationMethod = pInstrumentationMethod.Detach();
+    return S_OK;
+}
+
+// IProfilerManager2 Methods
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::DisableProfiling()
+{
+    return m_pProfilerManager->DisableProfiling();
+}
+
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::ApplyMetadata(_In_ IModuleInfo* pMethodInfo)
+{
+    return m_pProfilerManager->ApplyMetadata(pMethodInfo);
+}
+
+// IProfilerManager3 Methods
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::GetApiVersion(_Out_ DWORD* pApiVer)
+{
+    HRESULT hr = S_OK;
+    DWORD apiVer;
+    IfFailRet(m_pProfilerManager->GetApiVersion(&apiVer));
+    *pApiVer = apiVer;
+    return S_OK;
+}
+
+// IProfilerManager4 Methods
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::GetGlobalLoggingInstance(_Out_ IProfilerManagerLogging** ppLogging)
+{
+    HRESULT hr = S_OK;
+    CComPtr<IProfilerManagerLogging> pLogging;
+    IfFailRet(m_pProfilerManager->GetGlobalLoggingInstance(&pLogging));
+    *ppLogging = pLogging.Detach();
+    return S_OK;
+}
+
+
+// IProfilerManager5 Methods
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::IsInstrumentationMethodRegistered(_In_ REFGUID clsid, _Out_ BOOL* pfRegistered)
+{
+    HRESULT hr = S_OK;
+    BOOL fRegistered;
+    IfFailRet(m_pProfilerManager->IsInstrumentationMethodRegistered(clsid, &fRegistered));
+    *pfRegistered = fRegistered;
+    return S_OK;
+}
+
+// IProfilerManagerLogging Methods
+/*
+*/
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::LogMessage(_In_ const WCHAR* wszMessage)
+{
+    WCHAR wszPrefix[MAX_PATH];
+    wcscpy_s(wszPrefix, m_wszInstrumentationMethodGuid.c_str());
+    wcscat_s(wszPrefix, wszMessage);
```

Comment:
>wcscat_s [](start = 4, length = 8)

Did i miss something here? What if the message is bigger than MAX_PATH?

Reply:
FYI, if you look deeper into the CLogging/CLoggerService implementation, the buffer size is 4096.